src/transformers/generation/tf_utils.py

```diff
@@ -3857,58 +2901,11 @@ def sample_without_replacement(logits, num_samples):
     categorical sampling without replacement is currently not implemented the gumbel-max trick will do for now see
     https://github.com/tensorflow/tensorflow/issues/9260 for more info
     """
-    z = -tf.math.log(tf.random.uniform(shape_list(logits), 0, 1))
+    z = -tf.math.log(-tf.math.log(tf.random.uniform(shape_list(logits), 0, 1)))
```

Comment:
The solution for `sample_without_replacement` linked in its docstring (and corresponding references), aka gumbel-max trick, mentions "-log(-log(uniform(0, 1)))", i.e. apply the log twice.

Tagging @patrickvonplaten to double-check if this was a lapse, or if it was intended :)

Reply:
Ah yeah you're completely right! This is a big fix actually :-) Think this is by default used for TF sample

Reply:
Not sure if this has a big influence in practice, but this is definitely an important fix - thanks!